src/test/java/org/jabref/logic/net/URLDownloadTest.java

```diff
@@ -91,4 +95,24 @@ public void downloadOfHttpsSucceeds() throws IOException {
         Path path = ftp.toTemporaryFile();
         assertNotNull(path);
     }
+
+    @Test
+    public void testCheckConnectionSuccess() throws MalformedURLException {
+        URLDownload google = new URLDownload(new URL("http://www.google.com"));
+
+        int statusCode = google.checkConnection();
+        assertEquals(200, statusCode);
+    }
+
+    @Test
+    public void testCheckConnectionFail() throws MalformedURLException {
+        URLDownload nonsense = new URLDownload(new URL("http://nonsenseadddress"));
+        try {
+            int statusCode = nonsense.checkConnection();
+            fail();
```

Comment:
(obsolte with the refactoring to Boolean return value)

Why this statement? Can't the `try..catch` constrcut just be removed and a ` throws UnirestException` added to the method?

Please use AssertThrows for that. Details: https://howtodoinjava.com/junit5/expected-exception-example/